pyani_plus/workflows/snakemake_anim.smk

```diff
@@ -56,11 +56,21 @@ rule delta:
 # is used to calculate ANI values
 rule filter:
     params:
+        db=config["db"],
         delta_filter=config["delta_filter"],
         outdir=config["outdir"],
     input:
         "{outdir}/{genomeA}_vs_{genomeB}.delta",
+        genomeA=get_genomeA,
+        genomeB=get_genomeB,
     output:
         "{outdir}/{genomeA}_vs_{genomeB}.filter",
     shell:
-        "{params.delta_filter} -1 {input} > {wildcards.outdir}/{wildcards.genomeA}_vs_{wildcards.genomeB}.filter"
```

Comment:
Can't use ``{input}`` here anymore as that now expands to three filenames (the delta file and two FASTA files).

Reply:
I'm happy with the change if it works and make things faster/better. I'm just curious what change triggered this... just so I know for the future :)

Here `{input}` refered to `snakemake` input varaiable, in this case `{outdir}/{genomeA}_vs_{genomeB}.delta`,  so even if other files are present there, it shouldn't matter. Or I misunderstood something here.

Reply:
Originally there was only one input (the delta file).

Now there are three input files, the delta file (given to delta filter as before), plus the query and subject files (needed for the DB logging).

This is why ``{input}`` no longer gives just the delta filename, but three file names.